src/listing.rs

```diff
@@ -260,3 +306,39 @@ pub fn directory_listing<S>(
         .content_type("text/html; charset=utf-8")
         .body(html))
 }
+
+/// Converts a SystemTime object to a strings tuple (date, time)
+/// Date is formatted as %e %b, e.g. Jul 12
+/// Time is formatted as %R, e.g. 22:34
+///
+/// If no SystemTime was given, returns a tuple containing empty strings
+fn convert_to_utc(src_time: Option<SystemTime>) -> (String, String) {
+    match src_time {
+        Some(time) => {
+            let date_time = DateTime::<Utc>::from(time);
+            (
+                date_time.format("%e %b").to_string(),
+                date_time.format("%R").to_string(),
+            )
+        }
+        None => ("".to_string(), "".to_string()),
```

Comment:
Can't this be done more simply using `unwrap_or_default()` or such?

Reply:
I'll try :)